contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -165,13 +174,31 @@ abstract contract PoSValidatorManager is
         bool includeUptimeProof,
         uint32 messageIndex
     ) external {
+        if (!_initializeEndPoSValidation(validationID, includeUptimeProof, messageIndex)) {
+            revert ValidatorIneligibleForRewards();
+        }
+    }
+
+    function forceInitializeEndValidation(
+        bytes32 validationID,
+        bool includeUptimeProof,
+        uint32 messageIndex
+    ) external {
+        _initializeEndPoSValidation(validationID, includeUptimeProof, messageIndex);
+    }
+
+    function _initializeEndPoSValidation(
+        bytes32 validationID,
+        bool includeUptimeProof,
+        uint32 messageIndex
+    ) internal returns (bool) {
         PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
 
         Validator memory validator = _initializeEndValidation(validationID);
 
         // Non-PoS validators are required to boostrap the network, but are not eligible for rewards.
         if (!_isPoSValidator(validationID)) {
-            return;
+            return true;
```

Comment:
why do we return `true` here if below it returns for `reward > 0`? For non PoS validator shouldn't it return 0 rewards

Reply:
can we add unit tests to cover these cases, for example a non PoSvalidator that is claiming delegation fees

Reply:
This function returns `false` if 1) it's a PoS validator and 2) it's not able to claim rewards based on the submitted uptime. I switched this function to return a bool rather than the rewards, since in the call to `initializeEndValidation`, `0` rewards should revert for PoS validators, but not PoA.

Reply:
Unit testing this is not straightforward, since it requires a PoA -> PoS migration. I think this is better left for the e2e tests for now. In any case, I'm going to defer.

Reply:
We should be able to hit `!_isPoSValidator()` cases using the initial validator set

Reply:
Yeah to hit this case you should be able to do:
1. setup pos manager with 1 initial validator
2. add a new pos validator
3. remove the initial validator

Reply:
Thanks all. I've added a unit test that removes an initial validator to hit this case.